markdown/developers/pipeline_release_review_guidelines.md

```diff
@@ -0,0 +1,57 @@
+---
+title: Pipeline Release Review Guidelines
+subtitle: Suggestions for reviewing pipeline release PRs
+---
+
+Pipeline release reviews are often quite an overwhelming task, particularly when doing one for the first time.
+
+Pipeline release PRs are often quite diverse in their contents, making it harder to provide a strict set of reviewing guidelines that the PR must meet for approval, when compared to [modules](modules) or [subworkflows](subworkflows).
+
+Overall, the role of the revewer of pipeline release PRs is to check for adherence to the central principles of nf-core (reproducibility, execellent reporting, documented, keeping to the template etc.,). These are detailed in the pipeline guidelines [here](guidelines)
+
+Here we provide a general set of suggestions of what a reviewer should and do not need to look for when doing pipeline releases:
+
+## Do
+
+### nf-core principles
+
+- Does the pipeline meet the general [nf-core guidelines](guidelines)
+
+### Local code and modules
+
+- Do local scripts in `bin/` have author and license embedded?
+  - Local script licenses if written for the pipeline should not GPL, but ideally MIT (matching with pipeline code, upon which is assumed if not otherwise specified)
+- Do all local modules have docker/singularity/conda declarations?
+- Do all local modules conda tool declarations have versions?
+- Do all local modules report versions (if applicable)?
+  - Simple modules with e.g. single `grep` operations not necessary
+  - It would be good to add with more complex operations such as `awk`
+- Should any local modules be in nf-core/modules?
+
+### Documentation
+
+- Documention is only on the nf-core website (not pointed to other non-nf-core standard places, e.g. not read the docs )
+- Is documentation sufficiently described (`usage.md`, `output.md`, `nextflow_schema.json`)?
+  - nextflow_schema.json: check if types are correct and that `default` and `enum` are used where applicable
```

Comment:
Might be out of scope for this PR, but I think we need to start checking that it works with tower. We have seen some pitfalls in sarek with `null` iirc and it is very annoying for the user then. I am not sure if this can be done with json validation already or so? Maybe we could take this to the maintainers meeting list, if others agree

Reply:
Yes, take it to the maintainers! I'm not aware of that issue precisely, but that does ring a bell with me seeing stuff in the run headers for funcscan/taxprofiler :thinking: